numba/core/ssa.py

```diff
@@ -82,6 +84,36 @@ def _fix_ssa_vars(blocks, varname, defmap):
     return newblocks
 
 
+def _remove_unneeded_phis(phimap):
+    """Remove unneeded PHis from the phimap
```

Comment:
```suggestion
    """Remove unneeded PHIs from the phimap
```

Reply:
It'd also be good to add a comment about why this is necessary and what the conditions are for removal.